yogstation/code/game/gamemodes/wizard/raginmages.dm

```diff
@@ -0,0 +1,144 @@
+/datum/game_mode/wizard/raginmages
+	name = "ragin' mages"
+	config_tag = "raginmages"
+	required_players = 40
+	announce_span = "userdanger"
+	announce_text = "There are many, many wizards attacking the station!\n\
+	<span class='danger'>Wizards</span>: Accomplish your objectives and cause utter catastrophe!\n\
+	<span class='notice'>Crew</span>: Try not to die..."
+	var/max_mages = 0
+	var/making_mage = 0
+	var/mages_made = 1
+	var/time_checked = 0
+	var/bullshit_mode = 0 // requested by hornygranny
+	var/time_check = 1500
+	var/spawn_delay_min = 500
+	var/spawn_delay_max = 700
+
+/datum/game_mode/wizard/raginmages/post_setup()
+	..()
+	var/playercount = 0
+	if(!max_mages && !bullshit_mode)
+		for(var/mob/living/player in GLOB.mob_list)
+			if(player.client && player.stat != 2)
+				playercount += 1
+			max_mages = round(playercount / 8)
+			if(max_mages > 20)
+				max_mages = 20
+			if(max_mages < 1)
+				max_mages = 1
+	if(bullshit_mode)
+		max_mages = INFINITY
+
+/datum/game_mode/wizard/raginmages/check_finished()
+	var/wizards_alive = 0
+	for(var/datum/mind/wizard in wizards)
+		if(!istype(wizard.current,/mob/living/carbon))
+			continue
+		if(istype(wizard.current,/mob/living/brain))
+			continue
+		if(wizard.current.stat==DEAD)
+			continue
+		if(wizard.current.stat==UNCONSCIOUS)
+			if(wizard.current.health < 0)
+				to_chat(wizard.current, "<font size='4'>The Space Wizard Federation is upset with your performance and have terminated your employment.</font>")
+				wizard.current.death()
+			continue
+		wizards_alive++
+	if(!time_checked)
+		time_checked = world.time
+	if(bullshit_mode)
+		if(world.time > time_checked + time_check)
+			max_mages = INFINITY
+			time_checked = world.time
+			make_more_mages()
+			return ..()
+	if (wizards_alive)
+		if(world.time > time_checked + time_check && (mages_made < max_mages))
+			time_checked = world.time
+			make_more_mages()
```

Comment:
https://github.com/yogstation13/Yogstation-TG/blob/b6e38edbd73134678e06e0a1c6f9c6511bc045c2/yogstation/code/game/gamemodes/wizard/raginmages.dm#L50-L59
Can you consolidate all of this into one ``make_more_mages()`` call?